plugins/UFPWriter/UFPWriter.py

```diff
@@ -60,5 +60,43 @@ def write(self, stream, nodes, mode = MeshWriter.OutputMode.BinaryMode):
         else:
             Logger.log("d", "Thumbnail not created, cannot save it")
 
+        #Store the material.
+        application = Application.getInstance()
+        machine_manager = application.getMachineManager()
+        global_stack = machine_manager.activeMachine
+        container_registry = application.getContainerRegistry()
+
+        archive.addContentType(extension="xml.fdm_material", mime_type="application/x-ultimaker-material-profile")
```

Comment:
If the content type was already added to the archive this call will throw an exception. Even though it shouldn't already have been added, you never know (defensive programming).